speech/google/cloud/speech/result.py

```diff
@@ -17,6 +17,53 @@
 from google.cloud.speech.alternative import Alternative
 
 
+class Result(object):
+    """Speech recognition result representation.
+    This is the object that comes back on sync or async requests
+    (but not streaming requests).
+
+    :type alternatives: list
+    :param alternatives: List of
+        :class:`~google.cloud.speech.alternative.Alternative`.
+    """
+    def __init__(self, alternatives):
+        self.alternatives = alternatives
+
+    @classmethod
+    def from_pb(cls, result):
+        """Factory: construct instance of ``SpeechRecognitionResult``.
+
+        :type result: :class:`~google.cloud.grpc.speech.v1beta1\
+                               .cloud_speech_pb2.StreamingRecognizeResult`
+        :param result: Instance of ``StreamingRecognizeResult`` protobuf.
+
+        :rtype: :class:`~google.cloud.speech.result.SpeechRecognitionResult`
+        :returns: Instance of ``SpeechRecognitionResult``.
+        """
+        alternatives = []
+        for alt in result.alternatives:
+            alternatives.append(Alternative.from_pb(alt))
+        return cls(alternatives=alternatives)
+
+    @property
+    def confidence(self):
+        """Return the confidence for the most probable alternative.
+
+        :rtype: float
+        :returns: Confidence value, between 0 and 1.
+        """
+        return self.alternatives[0].confidence
```

Comment:
You could check for the `isFinal` alternative and return that confidence otherwise return the top result's confidence.

See: https://cloud.google.com/speech/reference/rest/v1beta1/speech/syncrecognize#SpeechRecognitionAlternative